src/components/buttons/Button.jsx

```diff
@@ -62,45 +56,57 @@ export const BUTTON_TOGGLE: {
   BLUE: 'blue',
 };
 
+const SPINNER_SIZE_MAP = {
+  [BUTTON_SIZE.L]: SPINNER_SIZE.SMALL,
+  [BUTTON_SIZE.M]: SPINNER_SIZE.XSMALL,
+  [BUTTON_SIZE.S]: SPINNER_SIZE.XXSMALL,
+};
+
+const SPINNER_COLOR_MAP = {
+  [BUTTON_TYPE.SOLID]: SPINNER_COLOR.WHITE,
+  [BUTTON_TYPE.SOLID_INVERTED]: SPINNER_COLOR.BLACK,
+  [BUTTON_TYPE.SOLID_BLUE]: SPINNER_COLOR.WHITE,
+  [BUTTON_TYPE.SOLID_MINT]: SPINNER_COLOR.WHITE,
+  [BUTTON_TYPE.SOLID_LIGHT]: SPINNER_COLOR.BLACK,
+  [BUTTON_TYPE.OUTLINE]: SPINNER_COLOR.BLACK,
+  [BUTTON_TYPE.TRANSPARENT]: SPINNER_COLOR.BLACK,
+  [BUTTON_TYPE.TRANSPARENT_LIGHT]: SPINNER_COLOR.GRAY700,
+  [BUTTON_TYPE.TRANSPARENT_PEACH]: SPINNER_COLOR.PEACH700,
+  [BUTTON_TYPE.TRANSPARENT_MUSTARD]: SPINNER_COLOR.MUSTARD700,
+  [BUTTON_TYPE.TRANSPARENT_BLUE]: SPINNER_COLOR.BLUE700,
+  [BUTTON_TYPE.TRANSPARENT_INVERTED]: SPINNER_COLOR.WHITE,
+  [BUTTON_TYPE.FACEBOOK]: SPINNER_COLOR.WHITE,
+};
+
+type ButtonType =
+  | 'solid-light'
+  | 'outline'
+  | 'transparent'
+  | 'transparent-light'
+  | 'solid'
+  | 'solid-inverted'
+  | 'solid-blue'
+  | 'solid-mint'
+  | 'transparent-peach'
+  | 'transparent-mustard'
+  | 'transparent-blue'
+  | 'transparent-inverted'
+  | 'facebook';
+
+type ButtonToggleType = 'peach' | 'mustard' | 'blue';
```

Comment:
Pls drop it someday 🙏  I would even move it directly to ButtonType

Reply:
this is not as easy as moving it to button type. It behaves differently for different types (sometimes we respect 3 values, sometimes only one is correct). definitely something to be further discussed though but more aboout having 2 buttons instead of 1 which we effectively have inside button. 

![image](https://user-images.githubusercontent.com/8572321/120162841-1b104280-c1f9-11eb-8204-8be2c29a1fc5.png)